src/lib.rs

```diff
@@ -525,12 +525,33 @@ impl fmt::Debug for RsaPrivate {
     }
 }
 
-#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[non_exhaustive]
 pub enum KeyUse {
-    #[serde(rename = "sig")]
     Signing,
-    #[serde(rename = "enc")]
     Encryption,
+    Custom(String),
```

Comment:
I'd also like to sprinkle `&'a str` and whatnot around this code as part of the v0.4 release. After using this crate a bunch, all the allocation is starting to get to me!